lua/weapons/weapon_snowball.lua

```diff
@@ -0,0 +1,133 @@
+SWEP.PrintName = "Snowballs"	
+
+SWEP.Instructions = "Left click to throw a snowball\nRight click changes trail color"
+
+SWEP.ViewModel = "models/weapons/v_snowball.mdl"
+SWEP.WorldModel = "models/weapons/w_snowball.mdl" 
+
+SWEP.Slot = 4
+SWEP.SlotPos = 1
+
+SWEP.DrawAmmo = false
+SWEP.DrawCrosshair = false
+SWEP.ViewModelFOV = 68
+
+SWEP.AutoSwitchTo = false
+SWEP.HoldType = "grenade"
+
+SWEP.Category = "Snowballs"
+SWEP.Spawnable = true
+
+SWEP.Primary.ClipSize = 1
+
+SWEP.Primary.Automatic = true
+SWEP.ThrowSound = Sound("weapons/iceaxe/iceaxe_swing1.wav")
+SWEP.ReloadSound = Sound("weapons/weapon_snowball/crunch.ogg")
+
+local ti = os.date("%B", os.time())
+if ti == "December" then --only activate during December
+	hook.Add("PlayerChangeLocation", "ChristmasSnowballs", function(ply, loc, old) --auto equip the snowball when outside
+		if !IsValid(ply) then return end
+		
+		local locname = GetLocationNameByIndex(loc)
+
+		if locname == "Outside" or locname == "Golf" and !ply:HasWeapon("weapon_snowball") then
+			ply:Give("weapon_snowball")
+		end
+	end)
+end
+
+if SERVER then --network the player's new color
+	util.AddNetworkString("CLtoSVSnowballColor")
+	util.AddNetworkString("SVtoCLSnowballColor")
```

Comment:
Please use a player:SetNWVector (use CLtoSVSnowballColor to network it from the player with the weapon to the server, then call that function on the server) for the color state. The reason for this is because if a player joins after the color was  set, using a NWVector will handle sending the correct color for them transparently.